common/src/lib.rs

```diff
@@ -429,6 +431,8 @@ pub struct Mappings {
     pub kernel_slice_start: u64,
```

Comment:
I think this is a physical address, which is only accessible in the bootloader's identity mapping. Perhaps it would be good to clarify this by changing the type:

```suggestion
    pub kernel_slice_start: PhysAddr,
```